CMakeLists.txt

```diff
@@ -77,29 +77,37 @@ endif()
 #
 option(SAIL_BUILD_APPS "Build applications." ON)
 option(SAIL_BUILD_EXAMPLES "Build examples." ON)
+option(SAIL_BUILD_TESTS "Build tests." ON)
```

Comment:
Yes, there is an unfortunate discrepancy in the CMake world. Some CMake variables start with `CMAKE_`, some not. I don't like `BUILD_TESTING` and `BUILD_SHARED_LIBS` too, but eventually I decided **not to introduce more chaos** and not to introduce `SAIL_BUILD_TESTING` and `SAIL_BUILD_SHARED_LIBS`. CMake users (and me too) frequently expect to set `BUILD_SHARED_LIBS` to `OFF` to build a static library for any project, and they (and me too :smile:) frequently frustrate if the project introduces some other option for that. Moreover, `SAIL_BUILD_SHARED_LIBS` must be mutually synchronized with `BUILD_SHARED_LIBS` introducing even more chaos.

See https://cmake.org/cmake/help/latest/module/CTest.html and https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html for more.